src/apprt/gtk/menu.zig

```diff
@@ -0,0 +1,136 @@
+const std = @import("std");
+
+const c = @import("c.zig").c;
+const apprt = @import("../../apprt.zig");
+const App = @import("App.zig");
+const Window = @import("Window.zig");
+const Surface = @import("Surface.zig");
+const Builder = @import("Builder.zig");
+
+const gtk = @import("gtk");
+const gdk = @import("gdk");
+const gio = @import("gio");
+const gobject = @import("gobject");
+
+const log = std.log.scoped(.gtk_menu);
+
+pub fn Menu(
+    comptime T: type,
+    comptime variant: []const u8,
+    comptime style: enum { popover_menu, popover_menu_no_arrow, popover_menu_bar },
+) type {
+    return struct {
+        const Self = @This();
+        const MenuWidget = switch (style) {
+            .popover_menu => gtk.PopoverMenu,
+            .popover_menu_no_arrow => gtk.PopoverMenu,
+            .popover_menu_bar => gtk.PopoverMenuBar,
+        };
+
+        parent: *T,
+        menu_widget: *MenuWidget,
+
+        pub fn init(self: *Self) void {
+            const name = switch (T) {
+                Window => "window",
+                Surface => "surface",
+                else => unreachable,
+            };
+            const parent: *T = @alignCast(@fieldParentPtr(variant, self));
+
+            var builder = Builder.init("menu-" ++ name ++ "-" ++ variant, .ui);
+            defer builder.deinit();
+
+            const object = builder.getObject("menu") orelse unreachable;
```

Comment:
orelse unreachable is just .?

Reply:
yeah, i guess i just find `.?` ugly but it doesn't matter to me.